src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -319,197 +368,175 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
             })?;
 
         let ty_res = self.resolve_path(&path_root, TypeNS, module_id);
-        let ty_res = if let Some(res) = ty_res {
-            res
-        } else {
-            return if ns == Namespace::ValueNS {
-                self.variant_field(path_str, current_item, module_id)
-            } else {
-                Err(ResolutionFailure::NotResolved {
-                    module_id,
-                    partial_res: None,
-                    unresolved: path_root.into(),
-                }
-                .into())
-            };
-        };
-        let res = match ty_res {
-            Res::PrimTy(_) | Res::Def(DefKind::Mod, _) => {
-                if let Some((path, prim)) = is_primitive(&path_root, TypeNS) {
-                    let impls =
-                        primitive_impl(cx, &path).ok_or_else(|| ResolutionFailure::NotResolved {
-                            module_id,
-                            partial_res: Some(prim),
-                            unresolved: item_str.into(),
-                        })?;
-                    for &impl_ in impls {
-                        let link = cx
+        ty_res
+            .and_then(|ty_res| {
```

Comment:
I don't think this is any more clear than before. It also increases the indentation a ton, which I don't like, there's already issues with right drift.

Reply:
Agreed - also doesn't make much sense without the behaviour change, since the solution would lack the ... "symmetry" I was going for. I'll either try to figure out what can be done, or drop that 3 commit altogether, probably.